docs/migrating/migratingfrommvc5/migratingfrommvc5.rst

```diff
@@ -0,0 +1,258 @@
+
+.. image:: _static/contact-page.png
+
+In MVC5 and previous versions of ASP.NET, static content was hosted from the root of the web project, and was intermixed with server-side files. In MVC6, all static content files are hosted from the /wwwroot folder, so we will need to adjust where we are storing our static content files. For instance, we can copy the favicon.ico file from the root of the original project to the /wwwroot folder in the new project.
+
+The MVC5 project uses Bootstrap for its styling, with files stored in /Content and /Scripts and referenced in /Views/Shared/_Layout.cshtml. We could simply copy the bootstrap.js and bootstrap.css files from the old project to the /wwwroot folder in the new project, but there are better ways to handle these kinds of client-side library dependencies in ASP.NET 5.
+
+In our new project, we'll add support for Bootstrap (and other client-side libraries), but we'll do so using the new support for client-side build tooling using bower and grunt. First, add a new Bower JSON Configuration File to the project root, called bower.json. In its "dependencies" property, add bootstrap, jquery, jquery-validation, and jquery-validation-unobtrusive. Add new properties for these items to the "exportsOverride" property as well, so that the complete bower.json file looks like this:
+
+.. code-block:: javascript
+
+	{
+		"name": "NewMvc6Project",
+		"private": true,
+		"dependencies": {
+			"bootstrap": "3.0.0",
+			"jquery": "1.10.2",
+			"jquery-validation": "1.11.1",
+			"jquery-validation-unobtrusive": "3.2.2"
+		},
+		"exportsOverride": {
+			"bootstrap": {
+				"js": "dist/js/*.*",
+				"css": "dist/css/*.*",
+				"fonts": "dist/fonts/*.*"
+			},
+
+			"jquery": {
+				"": "jquery.{js,min.js,min.map}"
+			},
+			"jquery-validation": {
+				"": "jquery.validate.js"
+			},
+			"jquery-validation-unobtrusive": {
+				"": "jquery.validate.unobtrusive.{js,min.js}"
+			}
+		}
+	}
+
+Bower will automatically download the specified dependencies, but for now the files are not yet in the wwwroot folder, and so cannot be requested by the browser:
+
+.. image:: _static/project-structure-bower.png
+
+Next, we will configure Grunt to process these files and place them where we want them in the wwwroot folder. First, we need to make sure Grunt is installed locally for the project. This is accomplished using NPM, which is similar to Bower but requires a different configuration file, "package.json". Add a new NPM configuration file to the root of the project, called package.json. Add the "grunt" and "grunt-bower-task" items to the devDependencies property, as shown (you should get Intellisense as you type their names):
```

Comment:
We have switched from Grunt to Gulp, so unfortunately the Grunt content will need to be updated to be based on Gulp.